File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.rb
# @param {Integer[]} cost
# @return {Integer}
def min_cost_climbing_stairs(cost)
  n = cost.size
  dp = Array.new(n + 1, 0)

  (2..n).each do |i|
    dp[i] = [dp[i - 1] + cost[i - 1], dp[i - 2] + cost[i - 2]].min
  end
  dp[n]
end

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function(cost) {
    cost.push(0)
    console.log(cost)
    for (let i=cost.length-3; i>=0; i--) {
        // cost[i] = Math.min(cost[i]+cost[i+1], cost[i]+cost[i+2])
        cost[i] += Math.min(cost[i+1], cost[i+2]) // Above Line Simplified
    }
    return Math.min(cost[0], cost[1])
};